pandas/core/missing.py

```diff
@@ -293,8 +293,8 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
         new_y = terp(new_x)
     elif method == 'spline':
-        # GH #10633
-        if not order:
+        # GH #10633, #24014
+        if order is None or not (0 < order):
```

Comment:
`not (0 < order)` --> `order >= 0` ?

Reply:
Done.  I had written it the original way so that it would raise on `order=numpy.nan`.  After the simplification, `numpy.nan` won't trigger the exception, but I think that's a reasonable balance.

Reply:
Hmm...I see.  Then let's add a comment about that.

Reply:
Do you think we should catch the case where `order` is nan?  I can add a comment either way, as you suggest.

Reply:
Actually, you can use `pandas.isnull`, which would simply this i.e.

`if pandas.isnull(order) or order >= 0`

Reply:
Of course, that's a good suggestion.  I used `isna`, which is already imported in the module and would catch a case of `None`.

Reply:
Yep, that works, since `isna` and `isnull` are one in the same function.